### ISSUE TITLE
Rule LibraryCodeMustSpecifyReturnType is not reported

### ISSUE THREAD

**Issue by ajax-semenov-y** [detekt/detekt#6009]
## Expected Behavior
Detekt reports LibraryCodeMustSpecifyReturnType violations

## Observed Behavior
Detekt does not reports LibraryCodeMustSpecifyReturnType violations

## Steps to Reproduce
My build file setup:
```kotlin
plugins {
    kotlin("jvm") version "1.8.0"
    application
    id("io.gitlab.arturbosch.detekt") version "1.22.0"
}
// .. omitted ...
dependencies {
    testImplementation(kotlin("test"))

    detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.22.0")
    detektPlugins("io.gitlab.arturbosch.detekt:detekt-rules-libraries:1.22.0")
}
// .. omitted ...
```
My code: 
```kotlin
class ExampleService {

    fun calculateName(arg: String) = "hello $arg"

    fun calculateAge(arg: Int) = 123 + arg
}
```

When I run `./gradlew detekt` it reports: 

```
.../src/main/kotlin/my/svc/ExampleService.kt:3:1: Class ExampleService should not be public [LibraryEntitiesShouldNotBePublic]
.../src/main/kotlin/my/svc/ExampleService.kt:7:34: This expression contains a magic number. Consider defining it to a well named constant. [MagicNumber]

```

Rule `LibraryEntitiesShouldNotBePublic` from the same rule set is reported, but `LibraryCodeMustSpecifyReturnType` is not. 

## Context
I want to force the team to specify a return type explicitly. I don't want to use Kotlin strict mode because I need only a return type check. 

## Your Environment
* Version of detekt used: 1.22.0
* Version of Gradle used (if applicable): 7.4.2 , 8.1

**Comment by cortinico** [comment]
> Rule `LibraryEntitiesShouldNotBePublic` from the same rule set is reported, but `LibraryCodeMustSpecifyReturnType` is not.

That's because `LibraryCodeMustSpecifyReturnType` requires type resolution and you're running with `detekt` which is not running Type Resolution rules.

More on this here: https://detekt.dev/docs/gettingstarted/type-resolution/

**Comment by cortinico** [comment]
If I'm not mistaken, you can see those information on the `debug` logs. More on this here:
- https://github.com/detekt/detekt/issues/5334
